### DISCUSSION TITLE
Field.render overrides not working

### DISCUSSION THREAD

**Discussion by joelstrom** [solspace/craft-freeform#1117] [Questions & Answers] [answered]
### Describe the bug
Using field.render() with custom overrides according to the [documentation](https://docs.solspace.com/craft/freeform/v5/templates/objects/field/#override-classes-values) here does not seem to output any of the defined overrides, only the default attributes and values.

We are trying to update our old form template to work with the new Freeforms versions. We have a switch that outputs different styling and attributes for few specific field types, and at the end we have a general field.render() that works as a fallback for the rest of the field types. Previously it was being done like this:

```
            {{ field.render({
              inputAttributes: input_attributes,
              class: input_classes,
              labelClass: label_classes
            }) }}
```

### Steps to reproduce
**Expectation**: Outputting field overrides with the help of the example below adds a class attribute with value "freeform-input" etc. to the rendered field

{{ field.render({
class: "freeform-input",
labelClass: "freeform-label" ~ (field.required ? " freeform-required" : ""),
errorClass: "freeform-errors",
instructionsClass: "freeform-instructions",
instructionsBelowField: true,
overrideValue: "This is now the new default value",
inputAttributes: {
"data-field-id": field.id,
}
}) }}

**Reality**: Field gets output with only default attributes and values, similarly to how it would if one would simply use {{ field.render() }} without added overrides.

Craft CMS version
4.8.0

Plugin version
5.1.0

**Additional context**
Craft recently upgraded from Craft 3.9 -> 4.8
Freeforms recently upgraded from 3.13.35 > 5.1.0 following the [upgrade guide](https://docs.solspace.com/craft/freeform/v5/setup/updating-freeform-4/). Only note regarding field render was regarding "option.checked" which has been fixed and seems to work well!

Partially related issue: https://github.com/solspace/craft-freeform/issues/995

**Comment by kjmartens** [comment, marked as answer]
Hi @joelstrom,

Sorry about this... we are still making some updates to our documentation. You'll want to check out the Template Overrides documentation for this instead:
https://docs.solspace.com/craft/freeform/v5/templates/formatting/#template-overrides

You have a couple of options now:

### At Form level:
https://docs.solspace.com/craft/freeform/v5/templates/formatting/#full-example

``` twig
{{ freeform.form("myFormHandle", {
    attributes: {
        novalidate: true,
        row: {
            class: "row",
        },
    },
    fields: {
        "@global": {
            attributes: {
                input: { class: "input-class" },
                label: { class: "label-class" },
            },
        },
        "@dropdown": {
            attributes: {
                input: { "+class": "select fullwidth" },
            },
        },
        "@checkboxes, @multiple-select": {
            instructions: "Select all that apply.",
        },
        ":required": {
            attributes: {
                label: { "+class": "form-required" },
            },
        },
    },
}).render() }}
```

### At Field level:
https://docs.solspace.com/craft/freeform/v5/templates/formatting/#on-all-parts-of-the-fields

``` twig
{{ field.render({
    label: "My Field Label Override",
    instructions: "My field instructions override.",
    value: "my field value override",
    attributes: {
        container: {
            class: "container",
        },
        input: {
            novalidate: true,
            class: "input-element",
        },
        label: {
            class: "label",
        },
        instructions: {
            class: "instructions",
        },
        error: {
            class: "error-block",
        },
    }
}) }}
```

Please let me know if you have any further questions. 🙂

**Comment by joelstrom** [comment]
Thank you for the quick response! We should be able to proceed with this.
